transport_plugins/bled112/iotile_transport_bled112/bled112_cmd_co.py

```diff
@@ -0,0 +1,843 @@
+from iotile.core.utilities.async_tools import OperationManager, SharedLoop
+from iotile.core.exceptions import HardwareError
+from .tilebus import *
+from .bgapi_structures import process_gatt_service, process_attribute, process_read_handle, process_notification
+from .bgapi_structures import parse_characteristic_declaration
+from .async_packet import InternalTimeoutError, DeviceNotConfiguredError
+
+BGAPIPacket = namedtuple("BGAPIPacket", ["is_event", "command_class", "command", "payload"])
+
+
+class AsyncBLED112CommandProcessor(threading.Thread):
+    def __init__(self, stream, commands, stop_check_interval=0.01, loop=SharedLoop):
+        super(AsyncBLED112CommandProcessor, self).__init__()
+
+        self._stream = stream
+        self._commands = commands
+        self._logger = logging.getLogger(__name__)
+        self._logger.addHandler(logging.NullHandler())
+        self._current_context = None
+        self._current_callback = None
+        self._event_check_interval = stop_check_interval
+
+        self._loop = loop
+        self._asyncio_cmd_lock = loop.create_lock()
+        self.operations = OperationManager(loop=loop)
+
+    def run(self):
+        while True:
+            try:
+                self._process_events()
+                event = self._commands.get(timeout=self._event_check_interval)
+                if event is None:
+                    self._logger.info("Shutting down bled112 thread due to stop command")
+                    return
+
+                cmdargs, callback, _, context = event
+                cmd = cmdargs[0]
+
+                if len(cmdargs) > 0:
+                    args = cmdargs[1:]
+                else:
+                    args = []
+
+                self._current_context = context
+                self._current_callback = callback
+
+                if hasattr(self, cmd):
+                    res = getattr(self, cmd)(*args)
+                else:
+                    pass #FIXME: Log an error for an invalid command
+
+                inprogress = False
+
+                if len(res) == 2:
+                    result, retval = res
+                else:
+                    result, retval, inprogress = res
+
+                self._current_context = None
+                self._current_callback = None
+
+                result_obj = {}
+                result_obj['command'] = cmd
+                result_obj['result'] = bool(result)
+                result_obj['return_value'] = retval
+                result_obj['context'] = context
+
+                if callback and inprogress is not True:
+                    callback(result_obj)
+            except Empty:
+                pass
+            except:
+                self._logger.exception("Error executing command: %s", cmd)
+                raise
+
+    def _set_scan_parameters(self, interval=2100, window=2100, active=False):
+        """
+        Set the scan interval and window in units of ms and set whether active scanning is performed
```

Comment:
Style - single line comment should end quotes on same line, though this may be close to the char limit then?

Reply:
Fixed.